server/game/GameActions/SwapDiscardWithHandAction.js

```diff
@@ -0,0 +1,47 @@
+const CardGameAction = require('./CardGameAction');
+
+class SwapDiscardWithHandAction extends CardGameAction {
+    setDefaultProperties() {
+        this.discardCard = null;
+    }
+
+    update(context) {
+        this.applyProperties(
+            Object.assign(
+                { target: this.getDefaultTargets(context), discardCard: context.source },
+                this.propertyFactory(context)
+            )
+        );
+    }
+
+    setup() {
+        this.name = 'swapDiscardWithHand';
+        this.effectMsg = 'swap {0} from hand with {1} from the discard';
+        this.effectArgs = this.discardCard;
+        super.setup();
+    }
+
+    canAffect(card, context) {
+        return (
+            card.location === 'hand' &&
+            this.discardCard.location === 'discard' &&
+            super.canAffect(card, context)
+        );
+    }
+
+    getEvent(card, context) {
+        return super.createEvent('onSwapDiscardWithHand', { card: card, context: context }, () => {
+            let discardIndex = this.discardCard.controller.discard.indexOf(this.discardCard);
+            let cardIndex = card.controller.hand.indexOf(card);
+            console.log('DISCARD', discardIndex, 'CARD', cardIndex);
```

Comment:
Need to remove this `console.log` statement

Reply:
Oops, done!